bgpd/bgp_route.c

```diff
@@ -3307,17 +3307,37 @@ bgp_clear_stale_route (struct peer *peer, afi_t afi, safi_t safi)
   struct bgp_info *ri;
   struct bgp_table *table;
 
-  table = peer->bgp->rib[afi][safi];
+  if ( safi == SAFI_MPLS_VPN)
+    {
+      for (rn = bgp_table_top (peer->bgp->rib[afi][safi]); rn; rn = bgp_route_next (rn))
+        {
+          struct bgp_node *rm;
+          struct bgp_info *ri;
 
-  for (rn = bgp_table_top (table); rn; rn = bgp_route_next (rn))
+          /* look for neighbor in tables */
+          if ((table = rn->info) != NULL)
```

Comment:
ok I hate this type of construct.  Having = inside of an if statement instead of == causes me issues.  Can we just move the assignment of table to the line above and then test for != NULL?